analysis/analysis_utils.py

```diff
@@ -108,14 +111,277 @@ def __init__(self):
+                    print('%s = %.2f' % (header, val))
+            print('---------------------------------------')
+            psi_curves = []
+            phi_curves = []
+            print(search.get_results(0,10))
+            for line in results:
+                psi_curve, phi_curve = search.lightcurve(line)
+                psi_curves.append(np.array(psi_curve).flatten())
+                phi_curve = np.array(phi_curve).flatten()
+                phi_curve[phi_curve == 0.] = 99999999.
+                phi_curves.append(phi_curve)
+                if line.lh < likelihood_level:
+                    likelihood_limit = True
+                    break
+            keep_idx_results = pool.starmap_async(return_indices,
+                                                  zip(psi_curves, phi_curves,
+                                                      [j for j in range(len(psi_curves))]))
+            pool.close()
+            pool.join()
+            keep_idx_results = keep_idx_results.get()
+
+            if len(keep_idx_results[0]) < 3:
+                keep_idx_results = [(0, [-1], 0.)]
+
+            for result_on in range(len(psi_curves)):
+
+                if keep_idx_results[result_on][1][0] == -1:
+                    continue
+                elif len(keep_idx_results[result_on][1]) < 3:
+                    continue
+                elif keep_idx_results[result_on][2] < likelihood_level:
+                    continue
+                else:
+                    keep_idx = keep_idx_results[result_on][1]
+                    new_likelihood = keep_idx_results[result_on][2]
+                    keep['results'].append(results[result_on])
+                    keep['new_lh'].append(new_likelihood)
+                    stamps = search.sci_stamps(results[result_on], 10)
+                    stamp_arr = np.array([np.array(stamps[s_idx]) for s_idx in keep_idx])
+                    keep['stamps'].append(np.sum(stamp_arr, axis=0))
+                    keep['lc'].append((psi_curves[result_on]/phi_curves[result_on])[keep_idx])
+                    keep['times'].append(image_params['mjd'][keep_idx])
+            print(len(keep['results']))
+
+            if len(keep['results']) > 500000:
+                with open('%s/memory_error.txt' %
+                          (res_filepath), 'w') as f:
+                    f.write('In %i total results, %i were kept. Needs manual look.' %
+                            (res_num + chunk_size, len(keep['results'])))
+                memory_error = True
+                likelihood_limit = True
+
+            if res_num+chunk_size >= 4000000:
+                likelihood_level = 20.
+                with open('%s/overload_error.txt' %
+                          (res_filepath), 'w') as f:
+                    f.write('In %i total results, %i were kept. Likelihood level down to %f.' %
+                            (res_num + chunk_size, len(keep['results']), line.lh))
+
+            res_num += chunk_size
+
```

Comment:
Do you think we should filter here instead of passing the entire result of the while loop back out and then in to the filtering? The only thing I would be worried about then I guess would be the clustering because we would cluster on separate chunks and still get duplicate results in the same area of parameter space. Maybe we could redo this to stamp filter each chunk and then do a clustering after. What do you think?

Reply:
I don't think we should do the clustering here, because both grid_search and region_search require processing, but region_search does not always require clustering, so I think it makes the most sense to just have filtering as a separate function.